lib/rspec/matchers/built_in/be.rb

```diff
@@ -236,9 +247,22 @@ def private_predicate?
           end
         end
 
-        def predicate_matches?
-          method_name = actual.respond_to?(predicate) ? predicate : present_tense_predicate
-          @predicate_matches = actual.__send__(method_name, *@args, &@block)
+        if RSpec::Support::RubyFeatures.kw_args_supported?
+          binding.eval(<<-CODE, __FILE__, __LINE__)
+          def predicate_matches?
+            method_name = actual.respond_to?(predicate) ? predicate : present_tense_predicate
+            if @kwargs.empty?
+              @predicate_matches = actual.__send__(method_name, *@args, &@block)
+            else
```

Comment:
What fails if we remove this?
```ruby
k={}
def a(x)
end
def b(y, z: 1)
end
a(1, **k)
b(2, **k)
```
neither of those calls results in a warning 😕

Reply:
I see the failures, but I'm having hard times understanding it, especially that the ArgumentError isn't raised where it should.

Reply:
It doesn't cause a warning, it sends an extra parameter to some places not expecting it.

Reply:
e.g. splatting keyword arguments, creates keyword arguments, and nil is not `{}`

Reply:
> you need the master branch of rspec-core to avoid that warning

👍 that recent change slipped my mind.

Got it. So I've played around with it, and it seems that `be_predicate()` is sending `__send__(:predicate, *[], **{})` and `**{}` goes for a regular parameter as `{}` 🤦 

```ruby
def m(p)
  puts p
end

__send__(:m, *[], **{})
```
```
warning: Passing the keyword argument as the last hash parameter is deprecated
{}
```

I don't have enough words to express my grief.